### PR TITLE
vimPlugins: warn if alias is used inside overrides

### PR DESCRIPTION
**Pull request by ryantm**
This changes the vimPlugins structure to be like stage.nix and top-level/aliases.nix.

vimPlugins was breaking evaluation if config.allowAliases was set to false, and we wanted to make sure it is harder to do that in the future.

### CODE REVIEW

**Comment by ryantm** [comment]
cc @Mic92

File: pkgs/misc/vim-plugins/default.nix [hunk 1]
```diff
@@ -28,422 +18,16 @@ let
 #
 # If additional modifications to the build process are required,
 # use add an override to this file.
```
Comment (Mic92) [inline pkgs/misc/vim-plugins/default.nix]:
```diff
- # use add an override to this file. 
+ # add an override to this file. 
```

File: pkgs/misc/vim-plugins/default.nix [hunk 2]
```diff
@@ -28,422 +18,16 @@ let
...
+  inherit (darwin.apple_sdk.frameworks) Cocoa CoreFoundation CoreServices;
+  inherit buildVimPluginFrom2Nix;
+};
 
-  fzfWrapper = buildVimPluginFrom2Nix {
-    name = fzf.name;
-    src = fzf.src;
-    dependencies = [];
-  };
+toFix = lib.foldl' (lib.flip lib.extends) (self: generated) ([
```
[the remark this answers is not in the document]
Comment (ryantm) [reply pkgs/misc/vim-plugins/default.nix]: Yes, I think so. Seems excessive in this case. Though the way it is we could easily add a feature allowing people to add their own vimPlugins overlays.

File: pkgs/misc/vim-plugins/default.nix (hunk 1 above)
Comment (ryantm) [reply pkgs/misc/vim-plugins/default.nix]: fixed

**Comment by ryantm** [comment]
@Mic92 Okay, I simplified it.

File: pkgs/misc/vim-plugins/default.nix [hunk 3]
```diff
@@ -1,449 +1,30 @@
...
-    buildPhase = ''
-      substituteInPlace plugin/youcompleteme.vim \
-        --replace "'ycm_path_to_python_interpreter', '''" \
-        "'ycm_path_to_python_interpreter', '${python}/bin/python'"
+overriden = generated // (overrides generated);
 
-      rm -r third_party/ycmd
-      ln -s ${ycmd}/lib/ycmd third_party
-    '';
+aliases = lib.optionalAttrs (config.allowAliases or true) (import ./aliases.nix lib overriden);
```
Comment (Mic92) [inline pkgs/misc/vim-plugins/default.nix]: Thanks. Now it makes sense to me.

File: pkgs/misc/vim-plugins/default.nix [hunk 4]
```diff
@@ -1,449 +1,30 @@
...
-, substituteAll
-, languagetool
-, Cocoa, CoreFoundation, CoreServices
-}:
+{ callPackage, config, lib, stdenv, vimUtils, vim, darwin, llvmPackages }:
 
 let
 
-  _skim = skim;
+inherit (vimUtils.override {inherit vim;}) buildVimPluginFrom2Nix;
```
Comment (Mic92) [inline pkgs/misc/vim-plugins/default.nix]: Do we care about indentation in this file?

Comment (ryantm) [reply pkgs/misc/vim-plugins/default.nix]: @Mic92 fixed the indentation

**Comment by ryantm** [comment]
@Mic92 Do you approve of this as is? We should probably merge it quickly before someone wants to do something with vim plugins.

**Review by Mic92 — APPROVED** [review]

**Pull request merged**